### PR TITLE
Manage peer routes

### PR DESCRIPTION
**Pull request by jakolehm**
Usually peers in a different region have `InternalIP` that is not directly connectable from master or workers in other regions. This PR automatically routes (DNAT) these peer address kubelet ports (10250) via Weave overlay network.

- exec/logs will work even if master is in a different region
- metrics server can scrape kubelets in different regions

Had to refactor things quite a bit...

### CODE REVIEW

File: lib/flying_shuttle/mixins/logging.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def logger
       return @logger if @logger
 
       @logger = Logger.new(STDOUT)
-      @logger.progname = self.class.name.sub('PCO::', '')
+      @logger.progname = self.class.name.sub('FlyingShuttle::', '')
```
Comment (kke) [inline lib/flying_shuttle/mixins/logging.rb]: Ruby 2.5 has `String#delete_prefix`

Comment (kke) [reply lib/flying_shuttle/mixins/logging.rb]:
```suggestion
      @logger.progname = self.class.name.delete_prefix('FlyingShuttle::')
```

**Comment by jakolehm** [comment]
@jnummelin @kke PTAL

File: lib/flying_shuttle/peer_service.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require_relative "mixins/logging"
+
+module FlyingShuttle
+  class PeerService
+    include Logging
+
+    REGION_LABEL = FlyingShuttle::REGION_LABEL
```
Comment (kke) [inline lib/flying_shuttle/peer_service.rb]: `REGION_LABEL` would return the same thing without this replication

**Review by kke — APPROVED** [review]
Much magic, can't spot anything obviously wrong.

**Comment by jakolehm** [comment]
Switched to DNAT, less magic and works better. Need to fix specs...

File: lib/flying_shuttle/peer_service.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    # @param external_addresses [Array<String>]
+    # @return [Array<String>]
+    def update_peers(this_peer, peers, external_addresses)
+      peer_addresses = []
+      peers.each do |peer|
+        if peer.region == this_peer.region
+          address = peer.address_for('InternalIP')&.address
+          peer_addresses << address if address
+        else
+          address = peer.metadata.labels[EXTERNAL_ADDRESS_LABEL]
```
Comment (jnummelin) [inline lib/flying_shuttle/peer_service.rb]: Could also be something like:
```
address = peer.address_for('ExternalIP') || peer.metadata.labels[EXTERNAL_ADDRESS_LABEL]
```
benefit would be that it's not tied to Pharos as such

File: lib/flying_shuttle/route_service.rb [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+require_relative 'mixins/weave_helper'
+require_relative 'mixins/logging'
+
+module FlyingShuttle
+  class RouteService
+    include Logging
+    include WeaveHelper
+
+    REGION_LABEL = FlyingShuttle::REGION_LABEL
+    IPTABLES_DNAT = 'iptables %s -w -t nat -p tcp -d %s --dport 10250  -j DNAT --to-destination %s:10250'
```
Comment (jnummelin) [inline lib/flying_shuttle/route_service.rb]: The first `%s` is actually two things, it combines the action and chain name. maybe consider splitting that.

**Review by jnummelin — APPROVED** [review]
Couple minor comments, super cool implementation

**Comment by jnummelin** [comment]
Specs still need some attention, some real iptables calls seem to leak out

File: lib/flying_shuttle/peer_service.rb (hunk 3 above)
Comment (jakolehm) [reply lib/flying_shuttle/peer_service.rb]: Yeah, fixed.

File: lib/flying_shuttle/route_service.rb (hunk 4 above)
Comment (jakolehm) [reply lib/flying_shuttle/route_service.rb]: I think I'm going to keep it as one.. but I'll refactor these as "named parameters" for easier reading.

**Review by jnummelin — APPROVED** [review]

**Pull request merged**